custom_components/multiscrape/service.py

```diff
@@ -151,7 +140,8 @@ async def _prepare_service_request(hass: HomeAssistant, conf, config_name):
     form_submit_config = conf.get(CONF_FORM_SUBMIT)
     parser = conf.get(CONF_PARSER)
     if form_submit_config:
-        form_http = create_http_wrapper(config_name, form_submit_config, hass, None)
+        form_http = create_http_wrapper(
+            config_name, form_submit_config, hass, None)
         form_submitter = create_form_submitter(
             config_name, form_submit_config, hass, form_http, None, parser
```

Comment:
You need to create a scraper and provide it here in case variables are configured. Did you test the service?

Reply:
Moved scraper creation to the `create_form_submitter` function, as it's better suited there.

Reply:
About testing the service... I did not... Testing it now, the example you put is not working anymore (even on master branch, and even the default configuration file) so I'm unsure what's working properly and what's not working properly in the service. Also, the coordinator is not initialized for sensor's under service, so I'm unsure how to handle form variables in the service at all. The logic I've added in service.py I currently made safe by checking if coordinator is initialized (which fixed an exception caused by that), but I can remove changes entirely in the service.py, whatever you prefer. Currently I do not need the service, working with form variables, so I'm fine with them not working. Waiting on your guidance. Maybe you fix the current aster branch and then I merge that and follow-up afterwards?
I would look into what's happening in more detail, but debugging does not work for me. The debugger attaches, but it does not hit breakpoints (it did when I initialy implemented the first version - header mappings). Any idea what's up with that?

Reply:
The example was not working because home-assistant.io was redesigned recently 😅
I updated the examples in the readme now.
I will also have a look at how to implement the variables in the service, as I find it important that there is feature parity.

Reply:
Thanks for fixing the README. You should update the example `configuration.yaml` as well.

Good news: I have service working for form variables! :)
The main issue is the custom handling of templates by multiscrape (`!` within `{{` and `}}`). Please look at the latest changes.

Reply:
Thanks for fixing the README. You should update the example `configuration.yaml` as well.

Good news: I have service working for form variables! :)
The main issue is the custom handling of templates by multiscrape (`!` within `{{` and `}}`). Please look at the latest changes.